src/test/objectstore/Fragmentation_simulator.cc

```diff
@@ -155,10 +161,211 @@ struct SimpleCWGenerator : public FragmentationSimulator::WorkloadGenerator {
 
     os->queue_transactions(ch, tls);
     os->verify_objects(ch);
+    tls.clear();
+    return 0;
+  }
+};
+
+typedef boost::mt11213b gen_type;
+
+struct RandomCWGenerator : public FragmentationSimulator::WorkloadGenerator {
+  std::string name() override { return "RandomCW"; }
+  int generate_txns(ObjectStore::CollectionHandle &ch,
+                    ObjectStoreImitator *os) override {
+
+    hobject_t h1;
+    h1.oid = fmt::format("obj1");
+    h1.set_hash(1);
+    h1.pool = 1;
+    ghobject_t obj1(h1);
+
+    hobject_t h2;
+    h2.oid = fmt::format("obj2");
+    h2.set_hash(2);
+    h2.pool = 1;
+    ghobject_t obj2(h2);
+
+    std::vector<ObjectStore::Transaction> tls;
+
+    ObjectStore::Transaction t1;
+    t1.create(ch->get_cid(), obj1);
+    tls.emplace_back(std::move(t1));
+
+    ObjectStore::Transaction t2;
+    t2.create(ch->get_cid(), obj2);
+    tls.emplace_back(std::move(t2));
+
+    os->queue_transactions(ch, tls);
+    os->verify_objects(ch);
+
+    gen_type rng(time(0));
+    boost::uniform_int<> u_size(0, _1Mb * 4);
+    boost::uniform_int<> u_offset(0, _1Mb);
+
+    for (unsigned i{0}; i < 200; ++i) {
+      tls.clear();
+
+      ObjectStore::Transaction t3;
+      auto size = u_size(rng);
+      auto offset = u_offset(rng);
+
+      t3.write(ch->get_cid(), obj1, offset, size, make_bl(size, 'c'));
+      tls.emplace_back(std::move(t3));
+
+      ObjectStore::Transaction t4;
+      size = u_size(rng);
+      offset = u_offset(rng);
+
+      t4.write(ch->get_cid(), obj2, offset, size, make_bl(size, 'c'));
+      tls.emplace_back(std::move(t4));
+
+      os->queue_transactions(ch, tls);
```

Comment:
queue_transactions does not provide any inherent kind of flow control.
Would it be run against bluestore, it would scheduled 200 ops.
It does not mean anything just yet, but will impact reusability of those tests against bluestore.

Reply:
I added `wait_till_finish` which will block until all in-flight txns are done